src/components/Counter.js

```diff
@@ -6,34 +6,42 @@ class Counter extends Component {
     incrementIfOdd = () => {
         // Stretch Problem: Implement an increment function that
         // only increments if the counter value is odd
+
+        if(this.props.count%2!==0){
+            return(
+                this.props.increment()
+            )
+        }
     };
 
     incrementAsync = () => {
         // Stretch Problem: Implement an increment function that
         // increments after waiting for one second
+        setTimeout(this.props.increment,1000)
     };
 
+
     render() {
         // Fill in the two button onClick methods
         // Upon clicking these buttons, the count
         // should decrement or increment accordingly
         return (
             <p>
                 Clicked: {this.props.count} times
-                <button onClick={() => {/* Fill me in */ }}>
+                <button onClick={()=>this.props.increment()}>
                     +
                 </button>
```

Comment:
You can also pass in the `increment` function directly - no need to use an inline anonymous function since you are not using any arguments anyway, like so: `onclick={this.props.increment}`